src/steps/resource-manager/key-vault/converters.test.ts

```diff
@@ -77,3 +78,44 @@ describe('createKeyVaultEntity', () => {
     });
   });
 });
+
+describe('createKeyVaultKeyEntity', () => {
+  const data: KeyProperties = {
+    id:
+      '/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev/providers/Microsoft.KeyVault/vaults/j1dev/keys/j1-key',
+    name: 'j1-key',
+    vaultUrl: 'https://j1dev.vault.azure.net/',
+    enabled: true,
+    expiresOn: new Date('2021-09-20T08:57:43.971Z'),
+  };
+
+  test('properties', () => {
+    const date = new Date('2021-09-20T08:57:43.971Z');
+
+    // this outputs object, 2021-09-20T08:57:43.971Z
+    console.log(typeof date, date);
+
+    expect(createKeyVaultKeyEntity(webLinker, data)).toEqual({
+      _class: ['Key'],
+      _key:
+        '/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev/providers/Microsoft.KeyVault/vaults/j1dev/keys/j1-key',
+      _rawData: [
+        {
+          name: 'default',
+          rawData: data,
+        },
+      ],
+      _type: 'azure_keyvault_key',
+      webLink:
+        'https://portal.azure.com/#@something.onmicrosoft.com/resource/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev/providers/Microsoft.KeyVault/vaults/j1dev/keys/j1-key',
+      name: 'j1-key',
+      enabled: true,
+      expiresOn: date,
```

Comment:
This field is coming out as the only mismatch during the comparison & hence causing the test to fail. Attaching a screenshot below for more context.
![image](https://user-images.githubusercontent.com/39617427/133978836-4d9ec98d-7da8-497d-856f-45e11ea4859d.png)

Reply:
I don't understand why writing `new Date()` isn't resulting in a `Date` object. Extremely weird behavior.